engine/src/main/java/org/terasology/world/block/family/FreeformFamily.java

```diff
@@ -75,7 +79,10 @@ public FreeformFamily(BlockFamilyDefinition blockFamilyDefinition, BlockBuilderH
 
 
     @Override
-    public Block getBlockForPlacement(Vector3i location, Side attachmentSide, Side direction) {
+    public Block getBlockForPlacement(Vector3i location, Side attachmentSide, Vector3fc viewingDirection) {
+        Vector3f attachmentSideNormal = new org.joml.Vector3f(attachmentSide.direction());
+        Side direction = ChunkMath.getSecondaryPlacementDirection(JomlUtil.from(viewingDirection), JomlUtil.from(attachmentSideNormal));
```

Comment:
This is a bit more clunky than what you've done `HorizontalFamily`. What's the difference?

```java
Side secondaryDirection = Side.inDirection(-viewingDirection.x(), 0, -viewingDirection.z());
```

As I cannot spot any other changes around the `Side direction` in both cases, shouldn't those two be the same?

Reply:
In `HorizontalFamily` I could improve and simplify the algorithm with the new concept. In this one I didn't want to touch the logic. So what you are seeing here is the exact same as it was originally, with the exception that `direction` is calculated within the method instead of being passed into it.

Reply:
Is the definition of the free form family to use the archetype or behave like the horizontal family?
I guess they could be using the same logic, but note that the calculation of `direction` works slightly different here than it currently does in `HorizontalFamily`.